src/services/persistent-storage.service.ts

```diff
@@ -0,0 +1,67 @@
+import { Promisify } from "../utils/types";
+import { defer } from "../utils/HelperFunctions";
+
+export class PersistentStorage implements Promisify<Storage> {
```

Comment:
The thing is that I'm trying to delay commiting the values to the store for when the stack is clear. Yes, it can be more complex but it eventually works the same as:
```
function whatever(){
     localStorage.setItem('...', '...')
}
```
but in the following syntax
```
async function whatever(){
      await persistentStorage.setItem('...', '...')
}
```

Reply:
The problem with async code is that it propagates through all the code that ultimately uses it so what was previously a simple sync function now becomes async (and so on up the stack). As we know Promises mean the potential of failure (`.catch`) so considerations are required to handle the catch (not entirely applicable here because local storage can throw synchronously but more in general). That's what I mean by added complexity